### ISSUE TITLE
Error while  putting document

### ISSUE THREAD

**Issue by damithc** [TEAMMATES/teammates#6847]
I've been getting a few of these errors recently from the live server. 
Here's an example:
```
Error Message: teammates.common.util.Logger severe: Operation did not succeed in time: putting document Document(documentId=abc@purdue.edu%CGT27208%2074176116, fields=[Field(name=courseId, value=CGT27208, type=TEXT), Field(name=searchableText, value=CGT27208,HCDD Learning Studio 3,abc@purdue.edu,ABC,Team 1,None, type=TEXT), Field(name=studentAttribute, value={
"googleId": "",
"email": "abc@purdue.edu",
"course": "CGT27208",
"name": "ABC",
"lastName": "DEF",
"comments": "",
"team": "Team 1",
"section": "None",
"key": "abc@purdue.edu%CGT27208%2074176116"
}, type=TEXT)], rank=195923046) into search index student.
```
v5.99

**Comment by damithc** [comment]
Getting a few more. Here's an example of another error that is cropping up frequently from the search side.
```
Error Message: teammates.common.util.Logger severe: Operation did not succeed in time: putting document Document(documentId=C5E0E5703B6983CB7C11A23788B1BEB2FC0B97C26575DF11C56A470A2036514A152F1F6503BA3100A38999C856xxxB069, fields=[Field(name=searchableText, value=ArqSoft_N7B_2017_s1,Software Architecture,XXX Calviño,xxx@gmail.com,,Co-owner,Instructor, type=TEXT), Field(name=instructorAttribute, value={
"courseId": "ArqSoft_N7B_2017_s1",
"name": "XXX Calviño",
"email": "xxx@gmail.com",
"isArchived": false,
"key": "xxx@gmail.com%ArqSoft_N7B_2017_s1577182779",
"role": "Co-owner",
"isDisplayedToStudents": true,
"displayedName": "Instructor",
"privileges": {
"courseLevel": {
"canmodifycourse": true,
"canmodifyinstructor": true,
"canmodifysession": true,
"canmodifystudent": true,
"canviewstudentinsection": true,
"canviewcommentinsection": true,
"cangivecommentinsection": true,
"canmodifycommentinsection": true,
"canviewsessioninsection": true,
"cansubmitsessioninsection": true,
"canmodifysessioncommentinsection": true
},
"sectionLevel": {},
"sessionLevel": {}
}
}, type=TEXT)], rank=195946586) into search index instructor.
```

**Comment by damithc** [comment]
OK, this may be due to this code in `SearchManager`.

```java
        if (elapsedTime >= Config.PERSISTENCE_CHECK_DURATION) {
            log.severe(String.format(ERROR_EXCEED_DURATION, document, indexName));
        }
```

I recently set the `PERSISTENCE_CHECK_DURATION` to `0`

**Comment by damithc** [comment]
@whipermr5 I think you know what to do here because you were dealing with a similar check in another place recently (i.e. we should not do the persistence check if the `PERSISTENCE_CHECK_DURATION` is `0`)

**Comment by damithc** [comment]
I've changed the persistence delay back to `1` (from `0`) but I still keep getting this error frequently, so I'm not sure if it was related to a recent change on our side or GAE side.

**Comment by damithc** [comment]
@whipermr5 can you check PRs merged for v5.99 to see if anything in those PRs can affect this issue. I've been getting quite a few of these errors last few days but when I rolled back to v5.98 the errors stopped (I need to monitor a while longer to be sure though).

**Comment by damithc** [comment]
> @whipermr5 can you check PRs merged for v5.99 to see if anything in those PRs can affect this issue. I've been getting quite a few of these errors last few days but when I rolled back to v5.98 the errors stopped (I need to monitor a while longer to be sure though).

Sorry @whipermr5 ignore this one. I found that persistence delay remained `0` even when I thought I reverted it back to previous value. So the problem should go away after this PR.
